cdap-cli/src/main/java/co/cask/cdap/cli/command/StartDebugProgramCommand.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.cli.command;
+
+import co.cask.cdap.cli.ArgumentName;
+import co.cask.cdap.cli.CLIConfig;
+import co.cask.cdap.cli.ElementType;
+import co.cask.cdap.cli.english.Article;
+import co.cask.cdap.cli.english.Fragment;
+import co.cask.cdap.cli.exception.CommandInputError;
+import co.cask.cdap.cli.util.AbstractAuthCommand;
+import co.cask.cdap.cli.util.ArgumentParser;
+import co.cask.cdap.client.ProgramClient;
+import co.cask.common.cli.Arguments;
+import com.google.gson.Gson;
+
+import java.io.PrintStream;
+import java.util.Map;
+
+/**
+ * Starts a program in debug mode.
+ */
+public class StartDebugProgramCommand extends AbstractAuthCommand {
+
+  private static final Gson GSON = new Gson();
+
+  private final ProgramClient programClient;
+  private final ElementType elementType;
+
+  public StartDebugProgramCommand(ElementType elementType, ProgramClient programClient, CLIConfig cliConfig) {
+    super(cliConfig);
+    this.elementType = elementType;
+    this.programClient = programClient;
+  }
+
+  @Override
+  public void perform(Arguments arguments, PrintStream output) throws Exception {
+    String[] programIdParts = arguments.get(elementType.getArgumentName().toString()).split("\\.");
+    if (programIdParts.length < 2) {
+      throw new CommandInputError(this);
+    }
+
+    String appId = programIdParts[0];
+    String programId = programIdParts[1];
+
+    String runtimeArgsString = arguments.get(ArgumentName.RUNTIME_ARGS.toString(), "");
+    if (runtimeArgsString == null || runtimeArgsString.isEmpty()) {
```

Comment:
I don't think we need to fetch the run time args and pass it. The preferences API would resolve the stored runtime arguments and pass it to the program. Can you open a JIRA for fixing this in 3.1, there are other places where this needs to change as well and not only for debug,